paperqa/core.py

```diff
@@ -68,12 +68,13 @@ async def map_fxn_summary(
     success = False
 
     if prompt_runner:
-        llm_result = await prompt_runner(
+        result = await prompt_runner(
             {"question": question, "citation": citation, "text": text.text}
             | (extra_prompt_data or {}),
             callbacks,
             "evidence:" + text.name,
         )
+        llm_result = result if isinstance(result, LLMResult) else result[0]
```

Comment:
Yeah the question is, what happens if 2+ results are returned? Wouldn't we just be throwing away useful data?

Can you make it safe by:

```python
if isinstance(result, Sequence) and len(result) != 1:
    raise NotImplementedError(f"We didn't handle blah blah {result} blah blah.")
llm_result = result if isinstance(result, LLMResult) else result[0]
```

In general, the design philosophy here is:

> It works, because it must work